neuralmagic/benchmarks/configs/benchmark_serving.json

```diff
@@ -0,0 +1,39 @@
+{
+	"configs": [
+		{
+			"description": "Benchmark vllm serving",
+			"dataset_download_cmds": [
+				"wget https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json"
+			],
+			"models": [
+				"facebook/opt-125m",
+				"PY007/TinyLlama-1.1B-step-50K-105b",
+				"mistralai/Mistral-7B-Instruct-v0.2",
+				"NousResearch/Llama-2-7b-chat-hf",
+				"mistralai/Mixtral-8x7B-Instruct-v0.1",
+				"codellama/CodeLlama-34b-hf",
+				"NousResearch/Llama-2-70b-chat-hf"
+			],
+			"sparsity": [],
+			"script_name": "benchmark_serving.py",
+			"script_args": {
```

Comment:
Do we think it makes sense for the script args to be `cross_producted` together?

I think that in most cases, we will primarily vary the `request_rate` with most other variables holding constant.

However, we may want to change `num_prompts` along with `request_rate` without having to run all the pairs so that each test terminates in a similar amount of time. E.g. if `request_rate=10`, 1200 prompts will take 2 minutes, but if `request_rate=1`, it will take 20 minutes. 

One way we could enable this is to not take the cross product, but instead parse the following to mean
```json
"script_args": {
  "num-prompts": [50,100,200,500],
  "request-rate": [0.5, 1.0, 2.5, 5.0],
  "best-of": [1],
  "dataset": ["sharegpt"]
}
```

4 runs:
- all use `best-of=1`
- all use `sharegpt`
- num_prompts,request_rate_pairs: `[(50,0.5), (100,1), (200,2), (500,5)]`


WDYT?

Reply:
I think it is too early to determine if doing the cross-product is required. I would suggest keeping it around. To handle the 
```
num_prompts,request_rate_pairs: [(50,0.5), (100,1), (200,2), (500,5)]
```
the better approach I believe is to add `--num-prompts-and-request-rate` as an argument to the benchmark_serving.py script. Ill make the change so we can see how it looks.

Reply:
Sounds good -> I like this approach better

Reply:
This is implemented 👍